docs/modules/extensions/pages/processors/block-macro-processor.adoc

```diff
@@ -0,0 +1,50 @@
+= Block Macro Processor
+
+BlockMacroProcessor is a SyntaxProcessor.
+
+== Registration method
```

Comment:
I like where you are going with this structure, but Registration method is only relevant when defining the extension using a Ruby block. If you define via a class, then we need to specify the class. So perhaps write this as:

```
== Registration

=== DSL method

block_macro

=== Parent class

BlockMacroProcessor
```

or something like that.

Reply:
My understanding is that you always need to call the appropriate registry method to add a processor to a registry.
I agree that explicitly mentioning the superclass is a good idea.